tests/returntypes/test_finite_difference_new.py

```diff
@@ -0,0 +1,959 @@
+        assert result[0].shape == (4,)
+        assert np.allclose(result[0], 0)
+
+        assert isinstance(result[1], numpy.ndarray)
+        assert result[1].shape == (4,)
+        assert np.allclose(result[1], 0)
+
+        assert isinstance(result[2], numpy.ndarray)
+        assert result[2].shape == (4,)
+        assert np.allclose(result[2], 0)
+
+        tapes, _ = qml.gradients.finite_diff_new(circuit.tape)
+        assert tapes == []
+
+    def test_all_zero_diff_methods_multiple_returns(self):
+        """Test that the transform works correctly when the diff method for every parameter is
+        identified to be 0, and that no tapes were generated."""
+
+        dev = qml.device("default.qubit", wires=4)
+
+        @qml.qnode(dev)
+        def circuit(params):
+            qml.Rot(*params, wires=0)
+            return qml.expval(qml.PauliZ(wires=2)), qml.probs([2, 3])
+
+        params = np.array([0.5, 0.5, 0.5], requires_grad=True)
+
+        result = qml.gradients.finite_diff_new(circuit)(params)
+
+        assert isinstance(result, tuple)
+
+        assert len(result) == 2
+
+        # First elem
+        assert len(result[0]) == 3
+
+        assert isinstance(result[0][0], numpy.ndarray)
+        assert result[0][0].shape == (1,)
+        assert np.allclose(result[0][0], 0)
+
+        assert isinstance(result[0][1], numpy.ndarray)
+        assert result[0][1].shape == (1,)
+        assert np.allclose(result[0][1], 0)
+
+        assert isinstance(result[0][2], numpy.ndarray)
+        assert result[0][2].shape == (1,)
+        assert np.allclose(result[0][2], 0)
+
+        # Second elem
+        assert len(result[0]) == 3
+
+        assert isinstance(result[1][0], numpy.ndarray)
+        assert result[1][0].shape == (4,)
+        assert np.allclose(result[1][0], 0)
+
+        assert isinstance(result[1][1], numpy.ndarray)
+        assert result[1][1].shape == (4,)
+        assert np.allclose(result[1][1], 0)
+
+        assert isinstance(result[1][2], numpy.ndarray)
+        assert result[1][2].shape == (4,)
+        assert np.allclose(result[1][2], 0)
+
+        tapes, _ = qml.gradients.finite_diff_new(circuit.tape)
+        assert tapes == []
+
+    def test_y0(self, mocker):
+        """Test that if first order finite differences is used, then
+        the tape is executed only once using the current parameter
+        values."""
+        dev = qml.device("default.qubit", wires=2)
+
+        with qml.tape.QuantumTape() as tape:
+            qml.RX(0.543, wires=[0])
+            qml.RY(-0.654, wires=[0])
+            qml.expval(qml.PauliZ(0))
+
+        tapes, fn = finite_diff_new(tape, approx_order=1)
+
+        # one tape per parameter, plus one global call
+        assert len(tapes) == tape.num_params + 1
+
+    def test_y0_provided(self):
+        """Test that if first order finite differences is used,
+        and the original tape output is provided, then
+        the tape is executed only once using the current parameter
```

Comment:
Is the device executed once here indeed? This part reads a bit confusing.

Reply:
It was copy pasted, I am rewriting it.